### ISSUE TITLE
Calendars are not retrieved on an iPad running iOS 17.1.

### ISSUE THREAD

**Issue by geertgeerits** [jfversluis/Plugin.Maui.CalendarStore#45]
Hi, 
Calendars are not retrieved on an iPad running iOS 17.1.

Visual Studio 2022 version 17.8.0 Preview 6.0 - .Net 8.0 and .Net 7.0 - Plugin.Maui.CalendarStore 1.02
macOS Sonoma 14.0 - Xcode 15.0.1

var calendars = await CalendarStore.Default.GetCalendars();
calendars.Count()  returns 0.

I have not been able to test it on an iPhone with iOS 17.1 because I do not have one available.

**Comment by jfversluis** [comment]
Looks like as of iOS 17 you will need to specify new permissions with more fine-grained levels to still access it: https://developer.apple.com/documentation/eventkit/accessing_the_event_store#2975207

**Comment by geertgeerits** [comment]
Thanks for your answer.
I had this in my Info.plist:
```
	<key>NSCalendarsUsageDescription</key>
	<string>This app wants to read from your calendars.</string>
```
I added this one:
```
	<key>NSCalendarsFullAccessUsageDescription</key>
	<string>This app wants to read from your calendars.</string>
```
	
But still the same problem.
If I delete the first one I get a message that I must add the NSCalendarsUsageDescription.

**Comment by jfversluis** [comment]
Yeah unfortunately that's not all that needs to be changed. I'll try to release an update soon. I have some code working locally already. I do think this change will be .NET 8 only unfortunately. Hope that's not a problem.

**Comment by geertgeerits** [comment]
Thank you, that's no problem for me. I have already switched to .Net 8.

**Comment by vikher** [comment]
I'm encountering the same issue where the calendars.Count() method consistently returns 0 when attempting to retrieve calendars using the Plugin.Maui.CalendarStore library in my development environment.

Environment Details:

Visual Studio for Mac: Version 17.6.7
.NET Version: 7.0
Plugin.Maui.CalendarStore: Version 1.02
macOS: Sonoma 14.0
Xcode: 15.0.1

Steps to Reproduce:

Attempt to retrieve calendars using CalendarStore.Default.GetCalendars().

1. Call calendars.Count() on the retrieved list of calendars.
2. Observe that calendars.Count() consistently returns 0, indicating no calendars are being retrieved.

Additional Information:

- I have verified that the necessary permissions to access calendars are in place, including the addition of NSCalendarsFullAccessUsageDescription in the app's Info.plist.
- Despite ensuring permissions and proper configuration, the issue persists.
- I have tested the application on an iPhone with iOS 17.1 

Thank you for your support and assistance in resolving this issue.

**Comment by jfversluis** [comment]
Sorry this took a while. I have just released v2.0-preview1 which should work again on iOS 17 as well as older versions. This does only support .NET 8 as the newer iOS APIs are only available there. Please let me know if there are any issues still with this.

Make sure to add the new keys to your info.plist as mentioned above.
